curation_concerns-models/app/actors/curation_concerns/file_set_actor.rb

```diff
@@ -103,11 +107,13 @@ def make_derivative(file_set_id, working_file)
         CharacterizeJob.perform_later(file_set_id, working_file)
       end
 
-      # @param [ActionDispatch::Http::UploadedFile] file
+      # @param [File, ActionDispatch::Http::UploadedFile] file
       # @param [String] id the identifer
       # @return [String] path of the working file
       def copy_file_to_working_directory(file, id)
-        copy_stream_to_working_directory(id, file.original_filename, file)
+        # file_set.label not gaurunteed to be set at this point (e.g. if called from update_content)
+        file_set.label ||= file.respond_to?(:original_filename) ? file.original_filename : ::File.basename(file)
```

Comment:
I see that it could get called from update_content, but in that case file_set never gets saved, so it's not required, right?